databus-connector/src/main/java/com/sequenceiq/cloudbreak/sigmadbus/processor/AbstractDatabusRecordProcessor.java

```diff
@@ -63,93 +57,88 @@ public void init() {
-    public void processRecord(DatabusRequest input) {
-        if (isDatabusProcessingEnabled()) {
-            try {
-                if (messageIsNotEmpty(input) && doesAccountIdExist(input)) {
-                    getProcessingQueues().process(input);
-                } else {
-                    LOGGER.debug("DataBusRecordInput needs both context with account ID and a payload message. Skip processing...");
-                }
-            } catch (InterruptedException e) {
-                LOGGER.debug("Putting DataBus put record request for processing is interrupted. {}", e.getMessage());
-            }
-        } else {
-            LOGGER.debug("Databus processing is not enabled, skip text message processing to databus");
-        }
+    @Override
+    public boolean isProcessingEnabled() {
+        return this.dataProcessingEnabled.get();
     }
 
     /**
-     * Override this to change the default behavior when an databus record processing error happens.
-     * Default behaviour: log the exception.
+     * Provide a valid (properly implemented) databus stream configuration.
      */
-    protected void handleDatabusRecordProcessingException(DatabusRequest input, DatabusRecordProcessingException e) {
-        LOGGER.warn(String.format("Exception during databus record processing [skip] - input: %s", input), e);
+    public C getDatabusStreamConfiguration() {
+        return this.databusStreamConfiguration;
     }
 
     /**
-     * Override this to change the default behavior when an unexpected error happens.
-     * Default behaviour: log the exception.
+     * Sigma dbus (grpc) related configuration
      */
-    protected void handleUnexpectedException(DatabusRequest input, Exception e) {
-        LOGGER.warn(String.format("Unexpected exception during databus record processing [skip] - input: %s", input), e);
+    public SigmaDatabusConfig getSigmaDatabusConfig() {
+        return this.sigmaDatabusConfig;
     }
 
     /**
-     * Override this to change the default behavior when the processing queue is too high.
-     * Default behaviour: log the dropped databus request input.
+     * Tracer
      */
-    protected void handleDroppedDatabusRequest(DatabusRequest input, int sizeLimit) {
-        LOGGER.warn("Blocking queue reached size limit: {}. Dropping databus input: {}", sizeLimit, input);
+    public Tracer getTracer() {
+        return tracer;
     }
 
     /**
-     * Round robin processing queue list that put data into the queue and record workers will pick those up.
+     * RegionAwareInternalCrnGeneratorFactory
      */
-    public RoundRobinDatabusProcessingQueues<C> getProcessingQueues() {
-        return processingQueuesRef.get();
+    public RegionAwareInternalCrnGeneratorFactory getRegionAwareInternalCrnGeneratorFactory() {
+        return regionAwareInternalCrnGeneratorFactory;
     }
 
     /**
-     * Checks that data processing is enabled for databus. Disabled if there are no databus workers.
+     * Service name
```

Comment:
this can be a bit more detailed -  in order to point out this is a service name is used as an identifier for dbus data processing or something similar